Mapify/Map/MapLifeCycle.cs

```diff
@@ -152,17 +202,62 @@ public static IEnumerator LoadMap(BasicMapInfo basicMapInfo)
             levelInfo.newCareerSpawnRotation = mapInfo.defaultSpawnRotation;
             levelInfo.enforceBoundary = true;
             levelInfo.worldBoundaryMargin = mapInfo.worldBoundaryMargin;
+        }
 
-            SetupStreamer(wsi.gameObject, mapInfo);
+        private static void ShowLoadingScreenImage(MapInfo mapInfo)
+        {
+            Mapify.Log("Showing custom loading screen image");
 
-            InitializeLists();
-            WorldStreamingInit_Awake_Patch.CanInitialize = true;
+            var randomInt = Random.Range(0, mapInfo.LoadingScreenImages.Length);
+            var customImage = mapInfo.LoadingScreenImages[randomInt];
 
-            foreach (VanillaAsset nonInstantiatableAsset in Enum.GetValues(typeof(VanillaAsset)).Cast<VanillaAsset>().Where(e => !AssetCopier.InstantiatableAssets.Contains(e)))
-                Mapify.LogError($"VanillaAsset {nonInstantiatableAsset} wasn't set in the AssetCopier! You MUST fix this!");
+            var canvasGameObject = Object.FindObjectsOfType<GameObject>().FirstOrDefault(gameObject => gameObject.name.Contains("LoadImage_Background_"));
```

Comment:
Is there no better way to find this? FindObjectsOfType can be *extremely* slow.

Reply:
I don't see how we can get rid of FindObjectsOfType entirely, but I did change it to search for RandomScreenPicker instead. That class is the only one that references the loading screen images.